pkgs/tools/networking/linux-router/default.nix

```diff
@@ -0,0 +1,83 @@
+{ stdenv, lib, fetchFromGitHub, makeWrapper
+
+# --- Runtime Dependencies ---
+, bash, procps, iproute2, dnsmasq, iptables
+, coreutils, flock, gawk, getopt, gnugrep, gnused, which
+# `nmcli` is not required for create_ap.
+# Use NetworkManager by default because it is very likely already present
+, useNetworkManager ? true
+, networkmanager
+
+# --- WiFi Hotspot Dependencies ---
+, useWifiDependencies ? true
+, hostapd, iw
+# You only need this if 'iw' can not recognize your adapter.
+, useWirelessTools ? true
+, wirelesstools # for iwconfig
+# To fall back to haveged if entropy is low.
+# Defaulting to false because not having it does not break things.
+# If it is really needed, warnings will be logged to journal.
+, useHaveged ? false
+, haveged
+# You only need this if you wish to show WiFi QR codes in terminal
+, useQrencode ? true
+, qrencode
+}:
+
+with lib;
+
+stdenv.mkDerivation rec {
```

Comment:
```suggestion
stdenv.mkDerivation rec {
```

Reply:
I would then to also change the following, is this preferable?

```diff
@@ -41,7 +39,7 @@ stdenv.mkDerivation rec {

   dontBuild = true;

-  installPhase = let
+  installPhase = with lib; let
       binPath = makeBinPath ([ procps iproute2 getopt bash dnsmasq
         iptables coreutils which flock gnugrep gnused gawk ]
         ++ optional useNetworkManager                          networkmanager
@@ -57,7 +55,7 @@ stdenv.mkDerivation rec {
       makeWrapper $out/.bin-wrapped/lnxrouter $out/bin/lnxrouter --prefix PATH : ${binPath}
     '';

-  meta = {
+  meta = with lib; {
     homepage = "https://github.com/garywill/linux-router";
     description = "Set Linux as router / Wifi hotspot / proxy in one command";
     longDescription = ''
```

Reply:
Yes and yes.